### ISSUE TITLE
Implement check=exists param to be passed to delete requests to throw error

### ISSUE THREAD

**Issue by kkanthet** [marklogic/node-client-api#407]
DELETE requests can now opt into a NOT FOUND error instead of idempotent silence when the document doesn't exist by passing a check=exists uri parameter

https://bugtrack.marklogic.com/44116

**Comment by wooldridge** [comment]
curl examples using new "check=exists" URL param...

**Document Does Not Exist:**

```
$ curl --anyauth --user admin:admin -X DELETE -i -H "Accept: application/json" \
"http://localhost:8558/v1/documents?uri=/nonexistent.json&check=exists"
HTTP/1.1 401 Unauthorized
Server: MarkLogic
WWW-Authenticate: Digest realm="public", qop="auth", 
nonce="eb1a8cd9b04e3df9bea6539c4383c5eb", opaque="03dbdcad4931ea03"
Content-Type: application/json; charset=utf-8
Content-Length: 104
Connection: Keep-Alive
Keep-Alive: timeout=5

HTTP/1.1 404 Not Found
Content-type: application/json; charset=UTF-8
Server: MarkLogic
Content-Length: 308
Connection: Keep-Alive
Keep-Alive: timeout=5

{"errorResponse":{"statusCode":404, "status":"Not Found", "messageCode":"XDMP-DOCNOTFOUND", 
"message":"XDMP-DOCNOTFOUND: xdmp:document-delete(\"/doc.json\", 
map:map(<map:map xmlns:xs=\"http://www.w3.org/2001/XMLSchema\" 
xmlns:xsi=\"http://www.w3.org/2001/XMLSchema-instance\" .../>)) 
-- Document not found"}}
```

**Document Exists:**

```
$ curl --anyauth --user admin:admin -X DELETE -i -H "Accept: application/json" \
"http://localhost:8558/v1/documents?uri=/doc.json&check=exists"
HTTP/1.1 401 Unauthorized
Server: MarkLogic
WWW-Authenticate: Digest realm="public", qop="auth", 
nonce="d259722fd0e5737e8649bbaf36871d7a", opaque="c2561d76d2a07e02"
Content-Type: application/json; charset=utf-8
Content-Length: 104
Connection: Keep-Alive
Keep-Alive: timeout=5

HTTP/1.1 204 Content Deleted
Server: MarkLogic
Content-Length: 0
Connection: Keep-Alive
Keep-Alive: timeout=5
```

**Comment by kcoleman-marklogic** [comment]
Sorry that you didn't find what you needed in the docs. I had it in there for 9.0-2, then had to pull it out, and haven't gotten around to sticking it back in. This was a good reminder though. :)

**Comment by wooldridge** [comment]
Product Management would like to wait on addressing this issue and do it as part of a broader update to CRUD semantics. Pushing out for now.

**Comment by kcoleman-marklogic** [comment]
so is this going in now? to REST as well?

**Comment by grechaw** [comment]
It's been in REST, and Java Client API uses it.  The "bug report" was for Java Client API and for some reason it was not considered necessary to document at REST layer (speaking from memory here, which is often faulty)

**Comment by kcoleman-marklogic** [comment]
It got implemented and then buried due to some controversy about underlying support in the server, which is why the documentation was pulled. Last I knew, that hadn't been resolved and was still on hold, which is why I was asking.

See internal bug 44116, whose last update still says: "Justin plans to handle this along with other RFEs that are similar (like document-insert options - not to upsert)."

**Comment by wooldridge** [comment]
This feature is not going into 2.1.1 for the Node.js Client API per Product Management's request.
